packages/frontend-2/components/developer-settings/CreateEditApplicationDialog.vue

```diff
@@ -0,0 +1,231 @@
+          show-required
+          :rules="[isItemSelected]"
+          show-label
+          :items="applicationScopes"
+          by="id"
+        />
+        <FormTextInput
+          v-model="redirectUrl"
+          label="Redirect URL"
+          help="After authentication, the users will be redirected (together with an access token) to this URL."
+          show-required
+          name="redirectUrl"
+          show-label
+          :rules="[isRequired, isUrl]"
+          type="text"
+        />
+        <FormTextInput
+          v-model="description"
+          label="Description"
+          help="A short description of your application."
+          name="description"
+          show-label
+          type="text"
+        />
+      </div>
+    </form>
+  </LayoutDialog>
+</template>
+
+<script setup lang="ts">
+import { useMutation } from '@vue/apollo-composable'
+import { AllScopes } from '@speckle/shared'
+import { LayoutDialog, FormSelectBadges } from '@speckle/ui-components'
+import {
+  ApplicationFormValues,
+  ApplicationItem
+} from '~~/lib/developer-settings/helpers/types'
+import {
+  createApplicationMutation,
+  editApplicationMutation
+} from '~~/lib/developer-settings/graphql/mutations'
+import { isItemSelected } from '~~/lib/common/helpers/validation'
+import { useForm } from 'vee-validate'
+import {
+  convertThrowIntoFetchResult,
+  getCacheId,
+  getFirstErrorMessage
+} from '~~/lib/common/helpers/graphql'
+import { useGlobalToast, ToastNotificationType } from '~~/lib/common/composables/toast'
+import { isRequired, isUrl, fullyResetForm } from '~~/lib/common/helpers/validation'
+
+const props = defineProps<{
+  application?: ApplicationItem
+}>()
+
+const emit = defineEmits<{
+  (e: 'application-created', applicationId: string): void
+}>()
+
+const isOpen = defineModel<boolean>('open', { required: true })
+
+const { mutate: createApplication } = useMutation(createApplicationMutation)
+const { mutate: editApplication } = useMutation(editApplicationMutation)
+const { triggerNotification } = useGlobalToast()
+const { handleSubmit, resetForm } = useForm<ApplicationFormValues>()
+
+const name = ref('')
+const scopes = ref<typeof applicationScopes.value>([])
+const redirectUrl = ref('')
+const description = ref('')
+
+const applicationScopes = computed(() => {
+  return Object.values(AllScopes).map((value) => ({
+    id: value,
+    text: value
+  }))
+})
+
+const onSubmit = handleSubmit(async (applicationFormValues) => {
+  const applicationId = props.application?.id
+
+  if (props.application) {
+    const result = await editApplication(
+      {
+        app: {
+          id: props.application.id,
+          name: name.value,
+          scopes: applicationFormValues.scopes.map((t) => t.id),
+          redirectUrl: redirectUrl.value,
+          description: description.value
+        }
+      },
+      {
+        update: (cache, { data }) => {
+          if (applicationId && data?.appUpdate) {
+            cache.modify({
```

Comment:
@fabis94 I was having trouble with this one. I've updated the query to include scopes.description, but no matter what I did to the cache update, I couldn't get it to work correctly, it always refetches all developer apps after an edit.

Maybe you could give me a call when you are back on this?

Reply:
@andrewwallacespeckle

1. I don't see any of your changes/attempts to fix it, maybe you forgot to push it? Code looks like it did at the beginning, so I had to try making the changes I described myself
2. The debugger made it easier to figure out what's going on. If you put a breakpoint on the `cache.modify` call you would be able to see that sometimes `applicationFormValues.name` is `undefined`, possibly because the name field hadn't been touched. Setting `name: () => undefined` caused Apollo to think that the field must be evicted entirely (?) and trigger the refetch. When making cache updates `undefined` isn't supported, its either supposed to be a real value or null.
3. You're using different values when actually making the GQL mutation and when making the cache mutation. You use `name.value` to specify the new name in the GQL request, but then using `applicationFormValues.name` in the cache.modify function. It's better to decide on the one to use and use that everywhere to avoid things. If you use both interchangeably one might be confused why `name.value` is set with a valid value and yet the cache.modify fn tries to set `undefined` instead (because of the different variable).

Reply:
I pushed in my fix: https://github.com/specklesystems/speckle-server/pull/1822/commits/74d051ca65ad3b16002e51697d0adbee78ba4b78

Reply:
Thanks for this. That makes sense re the undefined name, i've read over your commit and it all makes sense.